src/pytorch_lightning/utilities/rank_zero.py

```diff
@@ -97,3 +100,24 @@ class LightningDeprecationWarning(DeprecationWarning):
 
 
 rank_zero_deprecation = partial(rank_zero_warn, category=LightningDeprecationWarning)
+
+
+def _rank_prefixed_message(
+    message: str, trainer: Optional["pl.Trainer"] = None, log_rank_zero_only: bool = False
```

Comment:
it would require two functions:

```py
def get_rank(trainer)
   if trainer is not None:
        return trainer.global_rank
   else:
       return _get_rank()



def rank_prefixed_messasge(message, rank):
     return ...
```

And then in the log function:

```py
rank = get_rank(trainer)
msg = rank_prefixed_messasge(msg, rank)
if rank == 0:
    log(msg)
```

Reply:
Done in https://github.com/Lightning-AI/lightning/pull/14116/commits/00be095194637cf702975225d2b18a876376324f. Resolve if you are happy with that.

Reply:
Yes looks great